### PR TITLE
dkg/sync: add disconnect log to server

### PR DESCRIPTION
**Pull request by dB2510**
Add peer disconnected log to server.

category: misc
ticket: none

### CODE REVIEW

File: dkg/sync/server.go [hunk 1]
```diff
@@ -230,7 +230,9 @@ func (s *Server) Start(ctx context.Context) {
 
 		ctx = log.WithCtx(ctx, z.Str("peer", p2p.PeerName(stream.Conn().RemotePeer())))
 		err := s.handleStream(ctx, stream)
-		if err != nil {
+		if isRelayError(err) {
+			log.Warn(ctx, "Peer disconnected", nil)
+		} else if err != nil {
 			log.Warn(ctx, "Error serving sync protocol", err)
 		}
```
[the remark this answers is not in the document]
Comment (corverroos) [reply dkg/sync/server.go:237]: or maybe rather:
```
		if isRelayError(err) { // Relay errors are expected
			log.Debug(ctx, "Error serving sync protocol", nil)
		} else if err != nil {
			log.Warn(ctx, "Error serving sync protocol", err)
		}
```

**Review by xenowits — APPROVED** [review]

File: dkg/sync/client.go [hunk 2]
```diff
@@ -85,10 +85,10 @@ func (c *Client) Run(ctx context.Context) error {
 		relayBroke, connBroke, err := c.sendMsgs(ctx, stream)
 		c.clearConnected()
 		if relayBroke {
-			log.Debug(ctx, "Relay connection dropped, reconnecting")
+			log.Warn(ctx, "Relay connection dropped, reconnecting", nil, z.Str("peer", p2p.PeerName(stream.Conn().RemotePeer())))
```
Comment (corverroos) [inline dkg/sync/client.go:88]: so relay connections breaking is expected (it happens every 2min), we mostly ignore relay errors. Please revert this change.

File: dkg/sync/client.go [hunk 3]
```diff
@@ -85,10 +85,10 @@ func (c *Client) Run(ctx context.Context) error {
...
 			continue // Always reconnect on relay circuit recycling.
 		} else if connBroke && c.reconnect {
-			log.Debug(ctx, "Connection dropped, reconnecting")
+			log.Warn(ctx, "Connection dropped, reconnecting", nil, z.Str("peer", p2p.PeerName(stream.Conn().RemotePeer())))
```
Comment (corverroos) [inline dkg/sync/client.go:91]: peer name also added to ctx above. Not sure the stack traces of WARN level is useful, maybe just info level? Suggest: `log.Info(ctx, "Disconnected from peer")`

**Review by corverroos — CHANGES_REQUESTED** [review]

File: dkg/sync/client.go (hunk 3 above)
Comment (dB2510) [reply dkg/sync/client.go:91]: peer name added in ctx in local peer and not remote peer. Here's the screenshot with current WARN logs:
Screenshot 2022-12-14 at 12 19 10 PM

Comment (dB2510) [reply dkg/sync/client.go:91]: New logs:

Comment (corverroos) [reply dkg/sync/client.go]: when is the `WARN: Error serving sync protocol` logged?
when is the `ERRO: Fatal error...` logged?

Comment (corverroos) [reply dkg/sync/client.go]: The aim is to make it look nice:
 - Peer disconnecting isn't such a big problem, a simple "peer disconnected" info log is fine.
 - Relay connections are recycled every 2min, that shouldn't result in any logs (only simple debug) 
 - Warnings and Errors should only happen when things are crashing

**Review by corverroos — APPROVED** [review]
